node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/RetiredExpirer.java

```diff
@@ -1,69 +1,115 @@
 // Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
 package com.yahoo.vespa.hosted.provision.maintenance;
 
-import com.yahoo.collections.ListMap;
 import com.yahoo.config.provision.ApplicationId;
 import com.yahoo.config.provision.Deployer;
 import com.yahoo.config.provision.Deployment;
+import com.yahoo.config.provision.NodeType;
+import com.yahoo.vespa.applicationmodel.HostName;
 import com.yahoo.vespa.hosted.provision.Node;
 import com.yahoo.vespa.hosted.provision.NodeRepository;
 import com.yahoo.vespa.hosted.provision.node.History;
+import com.yahoo.vespa.orchestrator.OrchestrationException;
+import com.yahoo.vespa.orchestrator.Orchestrator;
 
 import java.time.Clock;
 import java.time.Duration;
+import java.time.Instant;
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
 import java.util.logging.Level;
+import java.util.stream.Collectors;
 
 /**
- * Maintenance job which deactivates nodes which has been retired.
- * This should take place after the system has been given sufficient time to migrate data to other nodes.
- * <p>
- * As these nodes are active, and therefore part of the configuration the impacted applications must be
- * reconfigured after inactivation.
+ * Maintenance job which deactivates retired nodes, if given permission by orchestrator, or
+ * after giving the system has been given sufficient time to migrate data to other nodes.
  *
- * @author bratseth
+ * @author hakon
  */
-public class RetiredExpirer extends Expirer {
+public class RetiredExpirer extends Maintainer {
 
-    private final NodeRepository nodeRepository;
     private final Deployer deployer;
+    private final Orchestrator orchestrator;
+    private final Duration retiredDuration;
+    private final Clock clock;
 
-    public RetiredExpirer(NodeRepository nodeRepository, Deployer deployer, Clock clock, 
-                          Duration retiredDuration, JobControl jobControl) {
-        super(Node.State.active, History.Event.Type.retired, nodeRepository, clock, retiredDuration, jobControl);
-        this.nodeRepository = nodeRepository;
+    public RetiredExpirer(NodeRepository nodeRepository,
+                          Orchestrator orchestrator,
+                          Deployer deployer,
+                          Clock clock,
+                          Duration maintenanceInterval,
+                          Duration retiredDuration,
```

Comment:
Nit: retiredDuration -> retiredExpiry to match the config (and it sounds a bit better)